src/main/java/com/hillel/items_exchange/controller/ImageController.java

```diff
@@ -0,0 +1,39 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.ImageDto;
+import com.hillel.items_exchange.service.ImageService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@Slf4j
+public class ImageController {
+    private final ImageService imageService;
+
+    @GetMapping("/adv/{advertisement_id}/product/{product_id}/image-urls")
+    public ResponseEntity<List<String>> getImageUrlsByAdvertisementIdAndProductId(
+            @PathVariable("advertisement_id") long advId,
+            @PathVariable("product_id") long productId) {
+        List<String> imageUrls = imageService.getImageUrlsByAdvertisementIdAndProductId(advId, productId);
+        return (imageUrls.isEmpty())
+                ? new ResponseEntity<>(HttpStatus.NOT_FOUND)
+                : new ResponseEntity<>(imageUrls, HttpStatus.OK);
+    }
+
+    @GetMapping("/adv/{advertisement_id}/product/{product_id}/images")
```

Comment:
The same is here - get rid of non-useful advertisement just make it as "/{product_id}"